docs/docs.md

```diff
@@ -0,0 +1,581 @@
+      match: string
+      attempts: number
+      successes: number
+      statName: string
+    }
+  | {
+      // 2018orwil
+      event: string
+      // qm1
+      match: string
+      attempted: boolean
+      succeeded: boolean
+      statName: string
+    })[]
+```
+
+# `/event/{eventKey}/teams`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = string[]
+```
+
+# `/events`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  name: string
+  district?: string
+  week?: number
+  // UTC date
+  startDate: string
+  // UTC date
+  endDate: string
+  location: {
+    lat: number
+    lon: number
+  }
+}[]
+```
+
+# `/schema`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  teleop: {
+    statName: string
+    statKey: string
+    type: "number" | "boolean"
+  }[]
+  auto: {
+    statName: string
+    statKey: string
+    type: "number" | "boolean"
+  }[]
+}
+```
+
+# `/team/{team}/automodes`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = string[]
+```
+
+# `/team/{team}/stats/auto`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  modeName: string
+  stats: (
+    | {
+        // 2018orwil
+        event: string
+        // qm1
+        match: string
+        attempts: number
+        successes: number
+        statName: string
+      }
+    | {
+        // 2018orwil
+        event: string
+        // qm1
+        match: string
+        attempted: boolean
+        succeeded: boolean
+        statName: string
+      })[]
+}[]
+```
+
+# `/team/{team}/stats/teleop`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  team: string
+  teleop: (
+    | {
+        attempts: {
+          max: number
+          avg: number
+        }
+        successes: {
+          max: number
+          avg: number
+        }
+        statName: string
+      }
+    | {
+        // total
+        attempts: number
+        // total
+        successes: number
+        statName: string
+      })[]
+  auto: (
+    | {
+        attempts: {
+          max: number
+          avg: number
+        }
+        successes: {
+          max: number
+          avg: number
+        }
+        statName: string
+      }
+    | {
+        // total
+        attempts: number
+        // total
+        successes: number
+        statName: string
+      })[]
+}
+```
+
+# `/users/{userId}/stars`
+
+## `GET`
+
+Anyone can view anyone's stars
+
+### Response
+
+```ts
+type Data = string[]
+```
+
+# `/users/{userId}`
+
+## `GET`
+
+Anyone can view any user
+
+### Response
+
+```ts
+type Data = {
+  username: string
+  name: string
+  admin?: true
+}
+```
+
+
+## `PUT`
+
+Only admins can modify other users
+
+### Request
+
+```ts
+type Data = {
+  username: string
+  name: string
+  password: string
+  admin?: boolean
+}
+```
+
+
+### Response
+
+```ts
+type Data = null
+```
+
+
+## `DELETE`
+
+Only admins can delete other users
+
+### Response
+
+```ts
+type Data = null
+```
+
+# `/users`
+
+## `POST`
+
+Anyone can create a user. For admins, the user will be created
```

Comment:
for admins the users will be verified automatically, for non-admins or non-authenticated users the user will not be verified and will require admin approval